app/lib/features/pins/pin_utils/pin_utils.dart

```diff
@@ -0,0 +1,167 @@
+import 'dart:io';
```

Comment:
this is quite complex stuff to follow. would be nice to have some integration tests for this to confirm it doesn't break (later)...?!?

Reply:
I planned to include tests right after its merging in addition of opening some important tickets to follow. Shall I include it in this PR or separate ?

Reply:
is there a github issue for it? then it's fine if that's done after.